functional/server_test.go

```diff
@@ -41,34 +41,60 @@ func TestReconfigureServer(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// NOTE: we need to sleep once here to get reliable results.
-	// Without this sleep, the entire fleetd test always ends up succeeding
-	// no matter whether SIGHUP came or not.
-	_, _ = cluster.MemberCommand(m0, "sh", "-c", `'sleep 2'`)
-
 	err = waitForFleetdSocket(cluster, m0)
 	if err != nil {
 		t.Fatalf("Failed to get a list of fleetd sockets: %v", err)
 	}
```

Comment:
Please submit another PR and make sure that it works, my change was to remove the sleep and make the test reliable so other PRs can be merged.